### PR TITLE
Cron

### PR DESCRIPTION
**Pull request by shreesha21**
Issue description: https://github.com/rancher/longhorn/issues/553

The following code changes include importing the Cron validation package and adding the cron validation for the input received from UI REST API and from storage class YAML files.

### CODE REVIEW

File: vendor.conf [hunk 1]
```diff
@@ -15,6 +15,7 @@ golang.org/x/sys                        98c5dad5d1a0e8a73845ecc8897d0bd56586511d
 gopkg.in/check.v1                       20d25e2
 github.com/satori/go.uuid               5bf94b69c6b68ee1b541973bb8e1144db23a194b
 gopkg.in/yaml.v2			v2.2.2
+github.com/robfig/cron                  b41be1df696709bb6395fe435af20370037c0b4c
```
Comment (yasker) [inline vendor.conf]: Needs to add the vendor directory as well. Drone failed.

File: csi/util.go [hunk 2]
```diff
@@ -64,6 +65,11 @@ func parseJSONRecurringJobs(jsonRecurringJobs string) ([]longhornclient.Recurrin
 	if err != nil {
 		return nil, fmt.Errorf("invalid json format of recurringJobs: %v  %v", jsonRecurringJobs, err)
 	}
+	for _, recurringJob := range recurringJobs {
+		if _, err := cron.ParseStandard(recurringJob.Cron); err != nil {
```
Comment (yasker) [inline csi/util.go]: Can it handle` recurringJob.Cron == ""`?

Comment (shreesha21) [reply csi/util.go]: Yes.

ParseStandard returns a new crontab schedule representing the given standardSpec (https://en.wikipedia.org/wiki/Cron). It differs from Parse requiring to always pass 5 entries representing: minute, hour, day of month, month and day of week, in that order. It returns a descriptive error if the spec is not valid.

It accepts

- Standard crontab specs, e.g. "* * * * ?"
- Descriptors, e.g. "@midnight", "@every 1h30m"

File: vendor.conf (hunk 1 above)
Comment (shreesha21) [reply vendor.conf]: Yes, will add and resend the PR

**Pull request merged**